test/promises/fail.js

```diff
@@ -0,0 +1,12 @@
+var test = require('../../');
+
+test('promise', function (t) {
```

Comment:
this test should be skipped when `typeof Promise === 'undefined'`

Reply:
I skipped them using node-tap in promise_fail.js, here https://github.com/substack/tape/pull/441/files#diff-0e44c4bcf1645d05ba9b5c2e44250c26R8

If that variable skip is truthy, then these test files aren't even run. (the builds that don't support promises show the skip in the output)

Should I switch to using test.skip in tape?

Reply:
hm - these tests need to continue to use tap, but it'd be ideal for the skip logic to be in each test file.

Reply:
It would produce a different (non-failing) TAP output then, which would fail the tap test instead of skipping it.

Should I put it in both? Maybe a comment explaining that they'll be skipped?

Reply:
ah yes, i like putting something in both - a generic "skip" mechanism in the tap test, and a specific one to promises here?

Reply:
Added a commit for this. It looks at the beginning of the stdout from the test, and if it's "skip", then it passes the test. Each test logs "skip" if `Promise` is undefined.

Is that what you had in mind?